string task_name
string[] adaptation_commands
